### PR TITLE
Add BFloat16 support for GLU, and randperm operators on CPU

### PR DESCRIPTION
**Pull request by CaoE**
add BFloat16 support for GLU and randperm operators on CPU

### CODE REVIEW

**Comment by saketh-are** [comment]
@VitalyFedyunin would you be able to review this?

**Comment by VitalyFedyunin** [comment]
yep, can do

**Review by VitalyFedyunin — COMMENTED** [review]
Code looks good to be, please include benchmarks comparing float and bfloat16 inputs of different sizes.

**Comment by CaoE** [comment]
Single core performance is tested on Xeon(R) Platinum 8180 @2.5 Ghz (AVX2).

Screenshot (120)

Single socket (28 cores) performance is tested on Xeon(R) Platinum 8180 @2.5 Ghz (AVX2).

Screenshot (121)

**Comment by CaoE** [comment]
Hi @VitalyFedyunin, could you please review it ? Thank you.

**Comment by frank-wei** [comment]
Same question in #63215

**Comment by CaoE** [comment]
Update:
The single core performance is tested on Xeon(R) Platinum 8180 @2.5 Ghz (AVX2).
Screenshot (300)


Single socket (28 cores) performance is tested on Xeon(R) Platinum 8180 @2.5 Ghz (AVX2).

Screenshot (303)

> Same question in #63215

As answered in https://github.com/pytorch/pytorch/pull/63215

**Comment by frank-wei** [comment]
TypeError: __init__() got an unexpected keyword argument 'dtypesIfCPU'. @mingfeima , when do you enable this type?

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -11418,6 +11418,7 @@ def ref_pairwise_distance(input1, input2):
            supports_autograd=True,
            sample_inputs_func=sample_inputs_glu,
            dtypes=floating_types(),
+           dtypesIfCPU=floating_types_and(torch.bfloat16),
```
Comment (mingfeima) [inline torch/testing/_internal/common_methods_invocations.py:11421]: 'dtypesIfCPU' keyword has been removed, just modify 'dtypes' will clear the CI failure.

`dtypes=floating_types_and(torch.bfloat16),`

Comment (CaoE) [reply torch/testing/_internal/common_methods_invocations.py:11421]: Thanks @mingfeima, I will fix and rebase it.

**Review by frank-wei — APPROVED** [review]
@pytorchbot  merge this

**Pull request closed without merging**